router/router.go

```diff
@@ -100,7 +100,9 @@ func (r *router) GetClient(key string) (interface{}, error) {
 
 	// no match so far, get a complete lock for creation
 	r.rw.Lock()
-	defer r.rw.Unlock()
+	defer func() {
```

Comment:
Good catch, this was part of my testing yesterday to see if I could test the 1 line that is in a race condition.

```go
defer func(){ ... }()
```

is the same as 

```go
defer r.rw.Unlock()
```

But instead of calling unlock on closing it calls the closure.